bentoml/server/bento_api_server.py

```diff
@@ -332,66 +368,66 @@ def bento_service_api_func_wrapper(self, api):
         service_version = self.bento_service.version
 
         def api_func_wrapper():
-            with self.metrics_request_in_progress.labels(
-                api_name=api.name, service_version=service_version
-            ).track_inprogress():
-
-                request_start_time = default_timer()
+            image_paths = []
+            if not config('logging').getboolean('disable_logging_image'):
+                image_paths = self.log_image(request, request_id)
 
-                image_paths = []
-                if not config('logging').getboolean('disable_logging_image'):
-                    image_paths = self.log_image(request, request_id)
+            # _request_to_json parses request as JSON; in case errors, it raises
+            # a 400 exception. (consider 4xx before 5xx.)
+            request_for_log = _request_to_json(request)
 
-                # _request_to_json parses request as JSON; in case errors, it raises
-                # a 400 exception. (consider 4xx before 5xx.)
-                request_for_log = _request_to_json(request)
+            # handle_request may raise 4xx or 5xx exception.
+            try:
+                response = api.handle_request(request)
+            except BentoMLException as e:
+                self.log_exception(sys.exc_info())
 
-                # handle_request may raise 4xx or 5xx exception.
-                try:
-                    response = api.handle_request(request)
-                except BentoMLException as e:
+                if 400 <= e.status_code < 500 and e.status_code not in (401, 403):
                     response = make_response(
                         jsonify(
                             message="BentoService error handling API request: %s"
                             % str(e)
                         ),
                         e.status_code,
                     )
+                else:
+                    response = make_response('', e.status_code)
+            except Exception:
+                # For all unexpected error, return 500 by default. For example,
+                # if users' model raises an error of division by zero.
+                self.log_exception(sys.exc_info())
 
-                request_log = {
-                    "request_id": request_id,
-                    "service_name": service_name,
-                    "service_version": service_version,
-                    "api": api.name,
-                    "request": request_for_log,
-                    "response_code": response.status_code,
-                }
+                response = make_response('', 500)
```

Comment:
add a general message here? "An error has occurred in BentoML user code when handling this request, find the error details in server logs"

Reply:
feel free to do in follow up PR, merging now 😄

Reply:
Ok